[Chang Jing Yan] iP

changjy44 opened the pull request:
### Duke
I have looked through your repository and it has come to my bewilderment that this duke program of yours is limited in its functionality.
It is beyond my pleasure to help you improve this duke program.🥰

Duke is a task tracker that helps you keep track of your tasks. Some functionalities include
- _Adding_ and _deletion_ of tasks.
- _Marking_ of tasks as done.

Running the program

1. Download the jar file from [here](https://github.com/changjy44/ip/releases/download/A-Jar/duke.jar).
2. Go to the directory that the jar file is located in. Ensure you have java installed.
3. Run
`java -jar duke.jar`

Supported commands include
- [ ] bye
- [ ] list
- [ ] mark
- [ ] unmark
- [ ] todo
- [ ] deadline
- [ ] event
- [ ] delete
- [ ] find

Please find the relevant document on how to use these commands.

> "This program solves world hunger" -- Confucius (idk when)

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World");
    }
}
```

Commit:
Add Gradle support

changjy44 committed:
Level-1

changjy44 committed:
Level-2

changjy44 committed:
Level-3

changjy44 committed:
Level-4: ToDos, Events and Deadlines

changjy44 committed:
A-TextUiTesting: Test using I/O redirection technique

changjy44 committed:
Level-5: Handle Errors for empty arguments after mark/tasks, validating number input after mark

changjy44 committed:
Level-6: Delete

changjy44 committed:
Enums

changjy44 committed:
Level-7: Save

changjy44 committed:
Merge branch 'Level-7.Save'

changjy44 committed:
Add Dates and TImes

changjy44 committed:
Merge branch 'Level-8-Dates-and-Times'

changjy44 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip into add-gradle-support

changjy44 committed:
A-OOP: Refactored Classes for Duke to include TaskList, Storage, Parser, and Commands

changjy44 committed:
A-Packages: Added duke, commands and tasks packages

changjy44 committed:
A-JUnit: Add test cases for Parser and TaskList classes

changjy44 committed:
A-Jar: Created a JAR file and tested its functionality

changjy44 committed:
A-JavaDoc: Added JavaDocs for all public classes, constructors and methods.

changjy44 committed:
Merge branch 'branch-A-JavaDoc'

changjy44 committed:
A-CodingStandard: Enforced all code to conform to the java coding standard

changjy44 committed:
Level-9: Added the find functionality to search for tasks with a matching keyword

changjy44 committed:
Merge branch 'branch-A-CodingStandard'

changjy44 committed:
Merge branch 'branch-Level-9'

jehousoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.Scanner;
+
+public class Duke {
+    private static final String FILE_PATH_DIR =  "data"; //"../../../data";
```
Could you maybe comment on the line before and indent based on the code?

jehousoh commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,12 @@
+package duke;
+
+public class DukeException extends Exception {
```
Could you add javadocs for this class?

jehousoh commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,18 @@
+package duke.commands;
+import duke.TaskList;
```
Could you add a space before the import?

jehousoh commented on src/main/java/duke/tasks/EventTask.java:
```diff
@@ -0,0 +1,47 @@
+        this.time = time;
+    }
+
+    public String getTime() {
```
Could you add a javadoc for this method?

jehousoh commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,60 @@
+    public boolean isMarked() {
+        return this.marked;
+    }
+    public String getName() {return this.name;}
```
could your return statement be on a separate line?

jehousoh reviewed (commented):
Looks good! 👍 
Just missing some minor changes but good job! All the best for the rest of the project~

bryanngzh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,95 @@
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
```
I like how you separated the import statements, perhaps you could also order them alphabetically as well?
Maybe you shouldn't import the whole duke library but just the ones you used (or have separate import statements) 🤔

bryanngzh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,95 @@
+
+        default:
+            throw new DukeException("No suitable name for that task");
+        }
```
Great indentation for the switch statements! 😄

bryanngzh commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,43 @@
+import duke.TaskList;
+import duke.Storage;
+import duke.DukeException;
+import java.time.LocalDateTime;
```
Could you perhaps add a space to separate the duke and java import statements? 🤔

bryanngzh commented on src/main/java/duke/tasks/DeadlinesTask.java:
```diff
@@ -0,0 +1,51 @@
+package duke.tasks;
+import java.lang.annotation.Inherited;
```
Could you perhaps have a spacing between import and package statements? 👍

bryanngzh reviewed (commented):
Overall, LGTM well done 👍 👍 😄 ! 
Just need to tidy up some formatting issues to ensure that your code is in line with the java writing conventions for CS2103T.
Keep up the great work!

changjy44 committed:
Added checkstyle features to the gradle project, and adjusted all code to conform with the style

changjy44 committed:
Merge branch 'branch-A-CheckStyle'

changjy44 committed:
Level-10: Added a graphical user interface for the application

changjy44 committed:
Merge branch 'branch-Level-10'

changjy44 committed:
A-Varargs: Added functionality to add multiple arguments for mark, unmark and delete.

changjy44 committed:
Merge branch 'branch-A-Varargs'

changjy44 committed:
Add welcome message upon initialisation.

The welcome message was missing upon adding the GUI.

Add welcome message through a method sayHello().

This allows the user to be greeted upon startup.

changjy44 committed:
Add assert statements.

The program have no assert statements which might cause the program to be buggy

Hence assert statements has been added following methods
parseLine() in Storage class
Add assert statements to check that the data is stored in the correct format of <type>,<mark>,<name>.

combineStringArray() in Parser class
Add assert statements to check that the start and the end parameters lie within the size of the array.

This allows for a more correct program by checking for both the preconditions.

changjy44 committed:
Merge pull request #1 from changjy44/branch-A-Assertions

Add assert statements.

changjy44 committed:
Improve code quality

changjy44 committed:
Merge pull request #2 from changjy44/branch-A-CodeQuality

Improve code quality

changjy44 committed:
Add streams to find command

changjy44 committed:
Merge pull request #3 from changjy44/branch-A-Streams

Add streams to find command

changjy44 committed:
Add Extension D-Expenses

changjy44 committed:
A-UserGuide

changjy44 committed:
A-Release
